pkg/xds/experimental/client/cell_test.go

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+package xdsclient
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/hivetest"
+
+	discoverypb "github.com/cilium/proxy/go/envoy/service/discovery/v3"
+	core_v1 "k8s.io/api/core/v1"
+)
+
+func TestCell_SuccessfullyRunClient(t *testing.T) {
+	var testXDSClient Client
+	hLog := hivetest.Logger(t)
+	h := hive.New(
+		node.LocalNodeStoreCell,
+		Cell,
+		cell.Invoke(func(localNodeStore *node.LocalNodeStore) {
+			localNodeStore.Update(func(n *node.LocalNode) {
+				hLog.Info("Update localNodeStore")
+				if n.Labels == nil {
+					n.Labels = make(map[string]string)
+				}
+				n.Labels[core_v1.LabelTopologyZone] = "zone"
+			})
+
+		}),
+		cell.Invoke(func(c Client) { testXDSClient = c }),
+	)
+	hive.AddConfigOverride(
+		h,
+		func(cfg *Config) {
+			cfg.ServerAddr = "dns:///fake-server.com:443"
+		})
+
+	if err := h.Populate(hLog); err != nil {
```

Comment:
nit: Start() implicitly does Populate()

Reply:
I wanted to populate the logger because I was checking the logs manually 😅